baseservice/iodevices/file_system/file_system_device_manager_base.py

```diff
@@ -12,68 +11,71 @@
 
 class FileSystemDeviceManagerBase:
     """
-    this manager is used to create FileSystem IO Devices
+    this manager is used to create IO Devices over a shared file system
     """
     DEFAULT_QUEUES_SUB_DIR = 'QUEUES'
+    DEFAULT_BOOKKEEPING_SUB_DIR = 'BOOKKEEPING'
+    DEFAULT_TMPDIR_SUB_DIR = 'TMP'
 
     def __init__(self,
                  root_folder: str,
                  queue_dir_name: str = DEFAULT_QUEUES_SUB_DIR,
-                 tmp_dir: Optional[str] = None,
-                 bookkeeping_path: Optional[str] = None,
+                 tmp_dir_name: str = DEFAULT_TMPDIR_SUB_DIR,
+                 bookkeeping_dir_name: str = DEFAULT_BOOKKEEPING_SUB_DIR,
                  serializer: Optional[FileSystemSerializerBase] = None):
         """
-        ctor
-
-        :param root_folder: the root folder to read/write from
+        :param root_folder: the root folder to use for the manager
         :param queue_dir_name: the name of the subdirectory under root_folder that holds the queues
-        :param tmp_dir: the full path of directory to use for temp files (None will generate a default under root_path)
+        :param tmp_dir_name: the name of the subdirectory under root_folder to use for temp files
+        :param bookkeeping_dir_name: the name of the subdirectory under root_folder that holds the book-keeping data
+        :param serializer: the serializer to use to write messages to files. None will use the default serializer
         """
         self._root_folder = root_folder
-        if tmp_dir is None:
-            tmp_dir = os.path.join(root_folder, 'TMP')
-        self._tmp_folder = tmp_dir
-        if bookkeeping_path is None:
-            bookkeeping_path = os.path.join(root_folder, 'BOOKKEEPING')
-
-        self._bookkeeping_folder = bookkeeping_path
-
         self._queues_folder = os.path.join(root_folder, queue_dir_name)
```

Comment:
I think that supporting pathlib's paths is both a modern requirement and will give you easier time implementing future functionalities

Reply:
example?